config/src/main/java/org/springframework/security/config/annotation/web/configurers/UrlAuthorizationConfigurer.java

```diff
@@ -49,7 +49,7 @@
  *
  * <pre>
  * protected void configure(HttpSecurity http) throws Exception {
- * 	http.apply(new UrlAuthorizationConfigurer&lt;HttpSecurity&gt;()).getRegistry()
+ * 	http.apply(new UrlAuthorizationConfigurer&lt;HttpSecurity&gt;(context)).getRegistry()
  * 			.requestMatchers(&quot;/users**&quot;, &quot;/sessions/**&quot;).hasRole(&quot;USER&quot;)
  * 			.requestMatchers(&quot;/signup&quot;).hasRole(&quot;ANONYMOUS&quot;).anyRequest().hasRole(&quot;USER&quot;);
  * }
```

Comment:
I think that instead of just fixing this, we should replace it with a `SecurityFilterChain` bean:

```java
@Bean
public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
http.apply(new UrlAuthorizationConfigurer&lt;HttpSecurity&gt;(context)).getRegistry()
 			.requestMatchers(&quot;/users**&quot;, &quot;/sessions/**&quot;).hasRole(&quot;USER&quot;)
 			.requestMatchers(&quot;/signup&quot;).hasRole(&quot;ANONYMOUS&quot;).anyRequest().hasRole(&quot;USER&quot;);
}
```

Reply:
Changed the documentation to define a bean as you suggested. Should I add the full definition as a Config class with @EnableWebSecurity and @Configuration annotation or will it suffice to define only as a Bean definition method?

Reply:
It can stay as it is, just the bean definition